internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/TestActiveActiveReplicationForIncPush.java

```diff
@@ -208,29 +217,97 @@ public void testAAReplicationForIncrementalPushToRT() throws Exception {
-        VeniceMultiClusterWrapper childDataCenter = childDatacenters.get(i);
-        // Verify the current version should be 1.
-        Version version =
-            childDataCenter.getRandomController().getVeniceAdmin().getStore(clusterName, storeName).getVersion(1);
-        Assert.assertNotNull(version, "Version 1 is not present for DC: " + dcNames[i]);
-      }
-      NativeReplicationTestUtils.verifyIncrementalPushData(childDatacenters, clusterName, storeName, 150, 2);
+  private void verifyForSeparateIncrementalPushTopic(
+      String storeName,
+      Properties propsInc1,
+      int dcIndexForSourceRegion) {
+    // Prepare TopicManagers
+    List<TopicManager> topicManagers = new ArrayList<>();
+    for (VeniceMultiClusterWrapper childDataCenter: childDatacenters) {
+      PubSubTopicRepository pubSubTopicRepository =
+          childDataCenter.getClusters().get(clusterNames[0]).getPubSubTopicRepository();
+      topicManagers.add(
+          IntegrationTestPushUtils
+              .getTopicManagerRepo(
+                  PUBSUB_OPERATION_TIMEOUT_MS_DEFAULT_VALUE,
+                  100,
+                  0l,
+                  childDataCenter.getKafkaBrokerWrapper(),
+                  pubSubTopicRepository)
+              .getLocalTopicManager());
+    }
+    // Run inc push with source fabric preference taking effect.
+    PubSubTopicPartition separateRealTimeTopicPartition = new PubSubTopicPartitionImpl(
+        PUB_SUB_TOPIC_REPOSITORY.getTopic(Version.composeSeparateRealTimeTopic(storeName)),
+        0);
+    PubSubTopicPartition realTimeTopicPartition =
+        new PubSubTopicPartitionImpl(PUB_SUB_TOPIC_REPOSITORY.getTopic(Version.composeRealTimeTopic(storeName)), 0);
+    try (VenicePushJob job = new VenicePushJob("Test push job incremental with NR + A/A from dc-2", propsInc1)) {
+      // TODO: Once server part separate topic ingestion logic is ready, we should avoid runAsync here and add extra
+      // check
+      CompletableFuture.runAsync(job::run);
+      TestUtils.waitForNonDeterministicAssertion(60, TimeUnit.SECONDS, true, () -> {
+        Assert.assertEquals(
+            job.getKafkaUrl(),
+            childDatacenters.get(dcIndexForSourceRegion).getKafkaBrokerWrapper().getAddress());
+        for (int dcIndex = 0; dcIndex < childDatacenters.size(); dcIndex++) {
+          long separateTopicOffset =
+              topicManagers.get(dcIndex).getLatestOffsetWithRetries(separateRealTimeTopicPartition, 3);
+          long realTimeTopicOffset = topicManagers.get(dcIndex).getLatestOffsetWithRetries(realTimeTopicPartition, 3);
+          // Real-time topic will have heartbeat messages, so the offset will be non-zero but smaller than the record
+          // count.
+          // DC 2 separeate real-time topic should get enough data.
+          if (dcIndex == dcIndexForSourceRegion) {
+            Assert.assertTrue(
+                separateTopicOffset > TestWriteUtils.DEFAULT_USER_DATA_RECORD_COUNT,
```

Comment:
In my opinion, we should not rely on record counts to verify whether data is being sent to the target topic. Instead, we could launch a consumer for the topic and check if the desired data has been pushed.

However, if for some reason we still want to rely on record counts, then we should consider disabling heartbeats to ensure accurate validation.